### PR TITLE
Raise an error when device or init_gpu_device is set and we can't use it for some reason.

### PR DESCRIPTION
**Pull request by abergeron**

### CODE REVIEW

File: theano/gpuarray/__init__.py [hunk 1]
```diff
@@ -173,26 +174,22 @@ def use(device,
...
-              config.init_gpu_device.startswith('opencl')):
-            if config.device != 'cpu':
-                raise ValueError(
-                    'you must set device=cpu to use init_gpu_device.')
-            if config.contexts != '':
-                print("Using contexts will make init_gpu_device act like device and move all computations by default, which might not be what you want.")
-            init_dev(config.init_gpu_device)
+    if (config.device.startswith('cuda') or
+            config.device.startswith('opencl')):
+        use(config.device)
```
Comment (nouiz) [inline theano/gpuarray/__init__.py]: This one still need to fall back. There is the flag force_device=True that make this raise an error.

Personaly, I'm fine of changing this: No possible fallback and remove force_device. But I would like @lamblin agreement before doing that.

Comment (abergeron) [reply theano/gpuarray/__init__.py]: I agree with you, we shouldn't (almost) silently ignore what the user asked for.

**Comment by abergeron** [comment]
jenkins test this please

**Comment by nouiz** [comment]
We don't want to change the interface now.

**Pull request closed without merging**